packages/contracts-bedrock/README.md

```diff
@@ -1,58 +1,232 @@
+        SystemConfig(<a href="https://github.com/ethereum-optimism/optimism/tree/develop/packages/contracts-bedrock/src/L1/SystemConfig.sol">SystemConfig</a>)
+    end
+
+    subgraph "User Interactions"
+        Users(Users)
+    end
+
+    subgraph "System Interactions"
+        Batcher(Batcher)
+        Proposer(Proposer)
+        Guardian(Guardian)
+    end
+
+    subgraph "Layer 2 Interactions"
+        L2Nodes(Layer 2 Nodes)
+    end
+
+    Batcher -->|publish transaction batches| BatchDataEOA
+    Proposer -->|propose state outputs| L2OutputOracle
+    Guardian -->|remove invalid state outputs| L2OutputOracle
+
+    ExternalERC20 <-->|mint/burn/transfer| L1StandardBridge
+    ExternalERC721 <-->|mint/burn/transfer| L1ERC721Bridge
+
+    L1StandardBridge <-->|send/receive message| L1CrossDomainMessenger
+    L1ERC721Bridge <-->|send/receive message| L1CrossDomainMessenger
+    L1CrossDomainMessenger <-->|package/send/receive message| OptimismPortal
+    L1StandardBridge -.->|query pause state| SuperchainConfig
+    L1ERC721Bridge -.->|query pause state| SuperchainConfig
+    L1CrossDomainMessenger -.->|query pause state| SuperchainConfig
+    OptimismPortal -.->|query pause state| SuperchainConfig
+
+    OptimismPortal -.->|query config| SystemConfig
+    OptimismPortal -.->|query proposed states| L2OutputOracle
+
+    Users <-->|deposit/withdraw ETH/ERC20| L1StandardBridge
+    Users <-->|deposit/withdraw ERC721| L1ERC721Bridge
+    Users -->|prove/execute withdrawal transactions| OptimismPortal
+
+    L2Nodes -.->|fetch transaction batches| BatchDataEOA
+    L2Nodes -.->|verify output roots| L2OutputOracle
+    L2Nodes -.->|fetch deposit events| OptimismPortal
+
+    classDef extContracts stroke:#ff9,stroke-width:2px;
+    classDef l1Contracts stroke:#bbf,stroke-width:2px;
+    classDef l1EOA stroke:#bbb,stroke-width:2px;
+    classDef userInt stroke:#f9a,stroke-width:2px;
+    classDef systemUser stroke:#f9a,stroke-width:2px;
+    classDef l2Nodes stroke:#333,stroke-width:2px
+    class ExternalERC20,ExternalERC721 extContracts;
+    class L1StandardBridge,L1ERC721Bridge,L1CrossDomainMessenger,L2OutputOracle,OptimismPortal,SuperchainConfig,SystemConfig l1Contracts;
+    class BatchDataEOA l1EOA;
+    class Users userInt;
+    class Batcher,Proposer,Guardian systemUser;
+    class L2Nodes l2Nodes;
+```
+
+#### Notes for Core L1 Smart Contracts
+
+- The `Batch Data Address` described above (**highlighted in GREY**) is *not* a smart contract and is instead simply an arbitrarily chosen account that is assumed to have no known private key. Generally, this account is chosen as the account `0xFF0000....<L2 chain id>` where `<L2 chain id>` is chain ID of the Layer 2 network for which the data is being posted. For instance, for OP Mainnet, this account is chosen as `0xFF00000000000000000000000000000000000010`.
```

Comment:
another naming nit, "batch inbox address" is what I've seen in the docs/config files. 

Also, tangential: interop specs [allow](https://specs.optimism.io/interop/dependency_set.html#chain-id) 32 byte chain IDs, which isn't compatible with this convention